lib/peer.go

```diff
@@ -183,14 +184,16 @@ func (pp *Peer) HandleGetTransactionsMsg(getTxnMsg *MsgDeSoGetTransactions) {
 		"MsgDeSoGetTransactions message with %v txns from peer %v",
 		len(getTxnMsg.HashList), pp)
 
-	mempoolTxs := []*MempoolTx{}
-	txnMap := pp.srv.mempool.readOnlyUniversalTransactionMap
+	mempoolTxs := []*MempoolTransaction{}
+
+	// We fetch the request txns from the PoS mempool because the PoS mempool contains
+	// all txns that have been added to the PoW and PoS mempools.
 	for _, txHash := range getTxnMsg.HashList {
-		mempoolTx, exists := txnMap[*txHash]
+		mempoolTx := pp.srv.posMempool.GetTransaction(txHash)
```

Comment:
We could use `pp.srv.GetMempool().GetTransaction(txHash)` and finish the `GetTransaction` implementation in the legacy mempool since both the legacy mempool and pos mempool implement the interface. My only concern about using the PoSMempool is if we start ejecting transactions from it when it fills up. `GetMempool` returns the legacy mempool if we're before the ProofOfStake2ConsensusCutoverBlockHeight block height.

Here's what the `GetTransaction` implementation could look like for the legacy mempool:
```
func (mp *DeSoMempool) GetTransaction(txnHash *BlockHash) *MempoolTransaction {
	mempoolTx, exists := mp.readOnlyUniversalTransactionMap[*txnHash]
	if !exists {
		return nil
	}
	return NewMempoolTransaction(mempoolTx.Tx, uint64(mempoolTx.Added.UnixMicro()))
}
```

Reply:
Given that it's okay if we only respond with a subset, my recommendation above is not necessary. but wanted you to think about it. I generally like how backward compatible the PoSMempool is so it's kinda nice to use it everywhere once a node has updated their code.